yaml-schema.json

```diff
@@ -0,0 +1,138 @@
+                                "type": "string",
+                                "description": "Pattern of the input channel"
+                            },
+                            "default": {
+                                "type": ["string", "number", "boolean", "array", "object"],
+                                "description": "Default value for the input channel"
+                            }
+                        },
+                        "required": ["type", "description"]
+                    }
+                }
+            }
+        },
+        "output": {
+            "type": "array",
+            "description": "Output channels for the module",
+            "items": {
+                "type": "object",
+                "patternProperties": {
+                    ".*": {
+                        "type": "object",
+                        "properties": {
+                            "type": {
+                                "type": "string",
+                                "description": "Type of the output channel"
+                            },
+                            "description": {
+                                "type": "string",
+                                "description": "Description of the output channel"
+                            },
+                            "pattern": {
+                                "type": "string",
+                                "description": "Pattern of the output channel"
+                            }
+                        },
+                        "required": ["type", "description"]
+                    }
+                }
+            }
+        },
+        "tools": {
+            "type": "array",
+            "description": "Tools used by the module",
+            "items": {
+                "type": "object",
+                "patternProperties": {
+                    ".*": {
+                        "type": "object",
+                        "properties": {
+                            "description": {
+                                "type": "string",
+                                "description": "Description of the output channel"
+                            },
+                            "homepage": {
+                                "type": "string",
+                                "description": "Homepage of the tool"
+                            },
+                            "documentation": {
+                                "type": "string",
+                                "description": "Documentation of the tool"
+                            },
+                            "tool_dev_url": {
+                                "type": "string",
+                                "description": "URL of the development version of the tool's documentation"
+                            },
+                            "doi": {
+                                "type": "string",
+                                "description": "DOI of the tool"
```

Comment:
Example pattern you could use: https://github.com/SPAAM-community/AncientMetagenomeDir/blob/230bca6d7af66b52c41fa79fc55cf51a4fdfb62d/ancientmetagenome-hostassociated/samples/ancientmetagenome-hostassociated_samples_schema.json#L53